cli/test/lib/exec/spawn_spec.js

```diff
@@ -154,6 +154,23 @@ describe('lib/exec/spawn', function () {
       })
     })
 
+    context('closes', function () {
+      ['close', 'exit'].forEach((event) => {
+        it(`if '${event}' event fired`, function () {
+          this.spawnedProcess.on.withArgs(event).yieldsAsync(0)
+
+          return spawn.start('--foo')
+        })
+      })
+
+      it(`if exit event fired and close event fired`, function () {
```

Comment:
no need for template string here, also I don't get what this test does. Does it call exit and close? where are any assertions

Reply:
it asserts that the Promise does not reject and it asserts that the timeout is not reached, which is what we're trying to test against

Reply:
and the result is that `exit` and `close` callbacks are called almost immediately, but still in order

Reply:
Using the guarantee that promises only ever resolve or reject once is such a overlooked but clutch use case that fits many different patterns